sql/sqlserver/server.go

```diff
@@ -333,3 +620,235 @@ func (s *Server) lookupDatabase(name string) (uint32, error) {
 	}
 	return uint32(gr.ValueInt()), nil
 }
+
+func (s *Server) processColumns(desc *structured.TableDescriptor,
+	node parser.Columns) ([]structured.ColumnDescriptor, error) {
+	if node == nil {
+		return desc.Columns, nil
+	}
+
+	cols := make([]structured.ColumnDescriptor, len(node))
+	for i, n := range node {
+		switch nt := n.(type) {
+		case *parser.StarExpr:
+			return s.processColumns(desc, nil)
+		case *parser.NonStarExpr:
+			switch et := nt.Expr.(type) {
+			case *parser.ColName:
+				// TODO(pmattis): If et.Qualifier is not empty, verify it matches the
+				// table name.
+				var err error
+				cols[i], err = findColumnByName(desc, et.Name)
```

Comment:
I had not rebased before sending out the PR. I'm about to send a new one that you guys will find easier to review. Thanks